lib/text/ui_text_displayer.js

```diff
@@ -382,15 +392,44 @@ shaka.text.UITextDisplayer = class {
       // TODO: Revisit this as we update the TTML parser.  See special case for
       // backgroundImage + region above.
       const percentageUnit = shaka.text.CueRegion.units.PERCENTAGE;
+
       const heightUnit = cue.region.heightUnits == percentageUnit ? '%' : 'px';
       const widthUnit = cue.region.widthUnits == percentageUnit ? '%' : 'px';
-      const viewportAnchorUnit =
-          cue.region.viewportAnchorUnits == percentageUnit ? '%' : 'px';
       style.height = cue.region.height + heightUnit;
       style.width = cue.region.width + widthUnit;
       style.position = 'absolute';
-      style.top = cue.region.viewportAnchorY + viewportAnchorUnit;
-      style.left = cue.region.viewportAnchorX + viewportAnchorUnit;
+
+      // Handle anchoring. Convert everything to percentages of the video
+      // container to avoid comparing percentages to pixels.
+      let width = cue.region.width;
+      let height = cue.region.height;
+      let viewportAnchorX = cue.region.viewportAnchorX;
+      let viewportAnchorY = cue.region.viewportAnchorY;
```

Comment:
Since these are converted below, I would prefer you move the variable definitions to right before the conversion.  That way, we know they are not referenced before that critical conversion happens.